### ISSUE TITLE
[Bug] Pass challenges with `raise`

### ISSUE THREAD

**Issue by F-park** [laike9m/Python-Type-Challenges#88]
## Bug description

![image](https://github.com/laike9m/Python-Type-Challenges/assets/52167622/5389cf14-8144-48af-9167-67626a5efb9f)

If `# expect-type-error` not in `test_code`, challenge can pass with `raise`.

## Challenges

- advanced-forward
- advanced-never2

**Comment by laike9m** [comment]
Yes, this is a known issue. I haven't thought of good ways to solve it

**Comment by laike9m** [comment]
Took another look. "advanced-forward" should be easy to fix, but never2 is hard, as I haven't found a way to force pyright to fail..

**Comment by F-park** [comment]
> But never2 is hard, as I haven't found a way to force pyright to fail..

`never2` can fix as below.

```py
## End of your code ##
from typing import Callable, assert_never, assert_type

assert_type(stop, Callable[[], Never])
assert_never(stop())
```

**Comment by laike9m** [comment]
Actually, I find that pyright will ignore whatever code following `raise`, since it thinks that those code is unreachable.

[Example](https://pyright-play.net/?strict=true&code=E4QwlgzgpgUDDEACADgT2GA5gCwC4C5EQA7EAG1QC8oBVU44ge1xFygBMAxAV2IGNcYRsQgBeAGbloMccEYBbRLlTIwxTIjDzkjYLiIRoegPrLksGGoj7RiALKoAwmRCGAFAA9RARgCUMVyNcUxUoNytcADo%2BRjQ3XwAaeycXQ38gA) 

So it really seems to have no solution..

**Comment by F-park** [comment]
> So it really seems to have no solution..

![image](https://github.com/laike9m/Python-Type-Challenges/assets/52167622/4fef3d08-c6fc-42fb-bf9b-c1d350e9ce34)

`Sourcery` can check the unreachable code, can we use it to solve this?

**Comment by laike9m** [comment]
Using sourcery is a no-go, using mypy maybe could be an option but certainly would complicate things. Another option is to create a special rule just for `raise` and warn users about it. I'll leave this one open and think about it.
